src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -332,7 +332,7 @@ public void Initialize(InitializeOptions options = InitializeOptions.None, DateT
 
                 _serverStore.StorageSpaceMonitor.Subscribe(this);
 
-                TaskExecutor.Execute( _ =>
+                ThreadPool.QueueUserWorkItem( _ =>
```

Comment:
This is here on purpose. 
The idea is that even if the Thread Pool is backed up, we have a dedicated thread that would still churn through those tasks.
We need to make sure that we aren't doing that twice, mind you.

Reply:
The task executor make a competition between the dedicated thread and a TP thread.

The problem that we had, is that the Task Executor run a never ending task, it got stuck, all operations completed by the TP, but the queue of the Task Executor exploded.

Reply:
So I left only non-blocking operations to be run by the Task Executor

Reply:
Okay, yes. `TaskExecuter`  is for very short notifications, yes.

Reply:
Like here,, no `NotifyFeaturesAboutStateChange` ?

Reply:
it is not short :-) 
1. It try to get a lock .. in a loop.
2. It handles every feature change..